auth/credentials/oauth_test.go

```diff
@@ -123,6 +124,31 @@ func TestNewServiceAccountOAuthClientWithTokenSource_WithWithExpiredToken(t *tes
 	assert.Equal(t, remoteToken, oAuthTokenSource.token.AccessToken)
 }
 
+// Test for providing valid Token and parsing the expiration date
+func TestNewServiceAccountOAuthClientWithTokenSource_WithWithAlmostExpiredToken(t *testing.T) {
+	// Generate a mock JWT Token that is going to expire in 5 seconds
+	expiration := time.Now().Add(-5 * time.Second)
```

Comment:
similar to previous comment, e.g. expirationAboutToHappen.


also these tests seems very similar, maybe we can have only one test func with multiple test cases like : 
``` 
	testCases := map[string]struct {
		expiration time.Time
		...
	}
             "almost expirer": ...,
	}

```

Reply:
SGTM. I have created jira to reduce duplication in tests.